File: services/shop-backend/src/main/java/io/miragon/shop/adapter/outbound/persistence/order/OrderPersistenceAdapter.java
package io.miragon.shop.adapter.outbound.persistence.order;

import io.miragon.shop.application.port.outbound.OrderRepository;
import io.miragon.shop.domain.order.Order;
import io.miragon.shop.domain.order.OrderId;
import io.miragon.shop.domain.shared.UserId;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class OrderPersistenceAdapter implements OrderRepository {

    private final OrderJpaRepository orderJpaRepository;

    public OrderPersistenceAdapter(OrderJpaRepository orderJpaRepository) {
        this.orderJpaRepository = orderJpaRepository;
    }

    @Override
    public Order save(Order order) {
        OrderEntity entity = OrderMapper.toEntity(order);
        OrderEntity saved = orderJpaRepository.save(entity);
        return OrderMapper.toDomain(saved);
    }

    @Override
    public Order loadById(OrderId orderId) {
        return orderJpaRepository.findById(orderId.getValue())
                .map(OrderMapper::toDomain)
                .orElse(null);
    }

    @Override
    public List<Order> loadById(UserId userId) {
        return orderJpaRepository.findByUserId(userId.getValue())
                .stream()
                .map(OrderMapper::toDomain)
                .toList();
    }
}

File: services/shop/shop-backend/src/main/kotlin/io/miragon/shop/adapter/outbound/persistence/order/OrderPersistenceAdapter.kt
package io.miragon.shop.adapter.outbound.persistence.order

import io.miragon.shop.adapter.outbound.persistence.order.OrderMapper.toDomain
import io.miragon.shop.adapter.outbound.persistence.order.OrderMapper.toEntity
import io.miragon.shop.application.port.outbound.OrderRepository
import io.miragon.shop.domain.order.Order
import io.miragon.shop.domain.order.OrderId
import io.miragon.shop.domain.shared.UserId
import org.springframework.stereotype.Component

@Component
class OrderPersistenceAdapter(
    private val orderJpaRepository: OrderJpaRepository
) : OrderRepository {

    override fun save(order: Order): Order {
        val entity = OrderEntity.Companion.toEntity(order)
        val savedEntity = orderJpaRepository.save(entity)
        return savedEntity.toDomain()
    }

    override fun loadById(orderId: OrderId): Order? {
        return orderJpaRepository.findById(orderId.value)
            .map { it.toDomain() }
            .orElse(null)
    }

    override fun loadById(userId: UserId): List<Order> {
        return orderJpaRepository.findByUserId(userId.value)
            .map { it.toDomain() }
    }
}